gcore/gdalpamdataset.cpp

```diff
@@ -916,29 +943,44 @@ CPLErr GDALPamDataset::TryLoadXML(char **papszSiblingFiles)
     /* -------------------------------------------------------------------- */
     /*      If we are looking for a subdataset, search for its subtree now. */
     /* -------------------------------------------------------------------- */
-    if (psTree && !psPam->osSubdatasetName.empty())
+    if (psTree)
     {
-        CPLXMLNode *psSubTree = psTree->psChild;
-
-        for (; psSubTree != nullptr; psSubTree = psSubTree->psNext)
+        std::string osSubNode;
+        std::string osSubNodeValue;
+        if (!psPam->osSubdatasetName.empty())
         {
-            if (psSubTree->eType != CXT_Element ||
-                !EQUAL(psSubTree->pszValue, "Subdataset"))
-                continue;
+            osSubNode = "Subdataset";
+            osSubNodeValue = psPam->osSubdatasetName;
+        }
+        else if (!psPam->osDerivedDatasetName.empty())
```

Comment:
the `else if` excludes the possibility to have both "Subdataset" and "DerivedDataset": are the options mutually exclusive?

Reply:
yes ad documented above in https://github.com/OSGeo/gdal/pull/8518/files#diff-8fa35e4c010f88e38395f7a251e702945e9000d65457f200dd5efe865a780ff8R790